app/scripts/proactive/view/FileBrowserView.js

```diff
@@ -126,53 +129,45 @@ define(
             }
             $.ajax({
                 url: that.dataspaceRestUrl + encodeURIComponent(pathname),
-                data: { "comp": "list", "includes": this.filterValue },
+                data: { "comp": "list-metadata", "includes": this.filterValue },
                 headers: { "sessionid": localStorage['pa.session'] },
-                async: false,
                 success: function (data){
-                    that.model['files'] = that.getFilesMetadata(data.fileListing);
-                    that.model['directories'] = that.getFilesMetadata(data.directoryListing);
+                    that.model['files'] = that.getFilesMetadata(data.fileListing, data);
+                    that.model['directories'] = that.getFilesMetadata(data.directoryListing, data);
                     that.model['showHidden'] = that.showHidden;
                     that.$el.html(that.template(that.model));
                     if(that.uploadRequest) {
                         that.switchToUploadingState();
                     }
+                    document.getElementById("filter-files-input").value = that.filterValue;
+                    new BeautifiedModalAdapter().beautifyForm(that.$el);
                 },
                 error: function (xhr, status, error) {
                     var errorMessage = "";
                     if(xhr) {
                         errorMessage = ": " + (xhr.status == 401 || xhr.status == 403 ? xhr.statusText : xhr.errorMessage);
                     }
                     StudioClient.alert('Error', "Failed to access " + pathname + errorMessage, 'error');
+                    document.getElementById("filter-files-input").value = that.filterValue;
+                    new BeautifiedModalAdapter().beautifyForm(that.$el);
```

Comment:
I propose to create a function for these two lines, and then call it in success and error